ext/openssl/ossl_pkey_ec.c

```diff
@@ -1021,6 +1021,101 @@ static VALUE ossl_ec_group_set_seed(VALUE self, VALUE seed)
 
 /* get/set curve GFp, GF2m */
 
+
+
+/*
+ * call-seq:
+ *   group.field_type   => Symbol or Integer
+ *
+ * See the OpenSSL documentation for EC_METHOD_get_field_type().
+ * If the field type is GFp (NID_X9_62_prime_field) it will return :GFp
+ * If the field type is GF2m (NID_X9_62_characteristic_two_field) it will return :GF2m
+ * If OpenSSL returns any other value it will return the raw NID_X9_62 value
+ */
+static VALUE ossl_ec_group_get_field_type(VALUE self)
+{
+    EC_GROUP *group;
+    GetECGroup(self, group);
+    if (!group) {
+        rb_raise(eEC_GROUP, "EC_GROUP is not initialized");
+    }
+
+    const EC_METHOD *method = EC_GROUP_method_of(group);
+    int field_type_id = EC_METHOD_get_field_type(method);
```

Comment:
Variable declarations need to be at the beginning of the block while we support Ruby <= 2.6.